eth-contracts/test/delegateManager.test.js

```diff
@@ -479,33 +435,42 @@ contract('DelegateManager', async (accounts) => {
         initialDelegateAmount,
         { from: delegatorAccount1 })
 
-      let total = 1000
-      let iterations = total
-      while (iterations > 0) {
-        let slash = false
-        if (((total - iterations) + 1) % 10 === 0) {
-          slash = true
-        }
-        console.log(`Round ${(total - iterations) + 1}`)
-        await fundClaimSlash(slash)
-        iterations--
-      }
+      // Fund new claim
+      await claimFactory.initiateRound()
+
+      // Get rewards
+      await delegateManager.claimRewards({ from: stakerAccount })
+      await delegateManager.claimRewards({ from: stakerAccount2 })
+
+      // Slash 30% of total
+      let slashNumerator = web3.utils.toBN(30)
+      let slashDenominator = web3.utils.toBN(100)
+      let totalInStakingContract = await staking.totalStakedFor(slasherAccount)
+      let slashAmount = (totalInStakingContract.mul(slashNumerator)).div(slashDenominator)
+
+      // Perform slash functions
+      await delegateManager.slash(slashAmount, slasherAccount)
 
       // Summarize after execution
       let spFactoryStake = await serviceProviderFactory.getServiceProviderStake(stakerAccount)
-      let totalInStakingContract = await staking.totalStakedFor(stakerAccount)
+      let totalInStakingAfterSlash = await staking.totalStakedFor(stakerAccount)
       let delegatedStake = await delegateManager.getTotalDelegatorStake(delegatorAccount1)
       let outsideStake = spFactoryStake.add(delegatedStake)
-      console.log(`SpFactory: ${spFactoryStake}, DelegateManager: ${delegatedStake}, Outside stake: ${outsideStake},  Staking: ${totalInStakingContract}`)
-      let stakeDiscrepancy = totalInStakingContract.sub(outsideStake)
-      console.log(`Stake discrepancy: ${stakeDiscrepancy}`)
-      let oneAud = toWei(1)
-      console.log(`1 AUD: ${oneAud}`)
-
-      let tokensInStaking = await token.balanceOf(stakingAddress)
-      console.log(`Tokens in staking ${tokensInStaking}`)
+      let stakeDiscrepancy = totalInStakingAfterSlash.sub(outsideStake)
+      let totalStaked = await staking.totalStaked()
+      let tokensAtStakingAddress = await token.balanceOf(stakingAddress)
+
+      assert.equal(stakeDiscrepancy, 0, 'Equal tokens expected inside/outside Staking')
+      assert.isTrue(totalStaked.eq(tokensAtStakingAddress), 'Expect equivalency between Staking contract and ERC')
+      assert.isTrue(totalInStakingAfterSlash.eq(outsideStake), 'Expected SP/delegatemanager to equal staking')
+      assert.isTrue((totalInStakingContract.sub(slashAmount)).eq(totalInStakingAfterSlash), 'Expected slash value')
     })
 
+    // TODO: Run claim / clash pattern 10,000x and confirm discrepancy
+    // TODO: What happens when someone delegates after a funding round has started...?
+    //        Do they still get rewards or not?
```

Comment:
What happens when someone delegates after a funding round has started but prior to the SP claiming rewards?
Do they still get rewards or not?
Potential idea - just lockup delegation for some interval
Potential idea 2 (heavier weight) - track each delegate's stake at a blocknumber similar to staking

Reply:
or lockup until after reward has been claimed (might be easiest)

Reply:
@roneilr lmk your thoughts on this behavior - i think disabling delegation prior to a claim is reasonable but would like your take